File: Python/Bubble_Sort.py
n = int(input("Enter number of elements:"))
print("Enter elements:")
array = []
for i in range(n):
    array.append(int(input()))
for i in range(n-1):
    for j in range(n-1-i):
        if array[j] > array[j+1]:
            temp = array[j]
            array[j] = array[j + 1]
            array[j + 1] = temp
print("Sorted array is:")
print(array)


"""
Problem : Sort the given array using bubble sort.

Traverse through all elements of array and in each pass place the element having maximum value at last of unsorted array. After each pass length of unsorted elements decreases hence we don't have to traverse all elements in every pass.

Sample I/O:

Enter number of elements:5
Enter elements:
5
4
3
2
1
Sorted array is:
[1, 2, 3, 4, 5]

Enter number of elements:4
Enter elements:
2
4
1
3
[1, 2, 3, 4]

Time complexity: O(n^2) (worst)
Space complexity: O(1)
"""

File: C#/BubbleSort.cs
using System;
					
public class Program
{
	public static void Main()
	{		
		int[] arrayInitial = { 100, 67, 95, 34, 1, 82, 23, 56, 74, 48, 99 };	
		Console.WriteLine("Array initial: " + Extensions.ArrayIntToString(arrayInitial));
		
		var sorted = new BubbleSort().Execute(arrayInitial);
		Console.WriteLine("Array sorted by Bubble Sort: " + Extensions.ArrayIntToString(sorted));
	}
}

public class BubbleSort
{
	public int[] Execute(int[] array)
	{
		int temp;
		
		foreach (int pos in array)
		{
			for (int i = 0; i <= array.Length - 2; i++) {
               if (array[i] > array[i + 1]) {
                  temp= array[i + 1];
                  array[i + 1] = array[i];
                  array[i] = temp;
               }
            }
		}	
		
		return array;
	}
}

public static class Extensions
{
	public static string ArrayIntToString(int[] array)
	{		
		var result = string.Empty;
		
		foreach (int pos in array)
			result += pos + " ";
		
		return result;
	}
}

/*
 Input used: 100, 67, 95, 34, 1, 82, 23, 56, 74, 48, 99
 
 Expected Output: Sorted Array by Bubble Sort: 1 23 34 48 56 67 74 82 95 99 100
 
 Time Complexity: O(n^2) 
 Space Complexity: O(1)

*/